### ISSUE TITLE
AD0001 (analyzer threw exception) diagnostics don't appear in error list

### ISSUE THREAD

**Issue by jasonmalinowski** [dotnet/roslyn#59214]
**Version Used**: 17.1 Preview 5

**Steps to Reproduce**:

1. Create an analyzer that throws exceptions. I just started with our default analyzer template that flags classes that don't have uppercase letters, and just added a throw in the symbol handler.
2. Use this from a project in VS>

**Expected Behavior**: I see live diagnostics for the exceptions being thrown.

**Actual Behavior**: I don't. A build does show diagnostics from the command build which then get reflected in the error list, but nothing seemingly live.

**Comment by mavasani** [comment]
Confirmed that this repros both with LSP pull diagnostics enabled and disabled. In fact, with LSP disabled, there does not seem to be anywhere that we report no location diagnostics (similar to AD0001 for analyzer exceptions). I think we should only attempt to fix this in the LSP pull diagnostics mode.

@dibarbet @CyrusNajmabadi - I think reporting no location diagnostics should go somewhere inside Workspace pull diagnostic handler, though all the diagnostic handlers and sources I looked at in the LSP side of code seem to be associated with documents. The most likely place seems to be here, but again that type and base type seem to be a document diagnostic source: https://sourceroslyn.io/#Microsoft.CodeAnalysis.LanguageServer.Protocol/Handler/Diagnostics/DiagnosticSources/WorkspaceDocumentDiagnosticSource.cs,30. We probably need a concept of project diagnostic source and figure out where to report such project level diagnostics without a source location.

**Comment by dibarbet** [comment]
@mavasani we should be reporting project diagnostics, but only when FSA is on - see https://sourceroslyn.io/#Microsoft.CodeAnalysis.LanguageServer.Protocol/Handler/Diagnostics/WorkspacePullDiagnosticHandler.cs,146 .  

Perhaps we should enable this for projects with open documents when FSA is off?  It seems like it would be a perf hit if we enabled it for all projects regardless of FSA.

**Comment by mavasani** [comment]
> Perhaps we should enable this for projects with open documents when FSA is off?

I don't think we want to enable whole project analysis even for projects with open documents when FSA is off. That will definitely be a perf hit. What we instead want is to report any project diagnostics that are reported by analyzers while analyzing open documents, without kicking off any additional analysis. This would need additional implementation in the OOP DiagnosticComputer side to keep track of any reported project diagnostics while analyzing documents, and also in the DiagnosticAnalyzerService to expose a new API to fetch existing project diagnostics from document analysis. Let me take up this work, and once there is a new API on IDiagnosticAnalyzerService for the same, I'll move the issue to you to hook these up on LSP handler side.
